### ISSUE TITLE
mconfig -s does not create statically linked binaries

### ISSUE THREAD

**Issue by DrDaveD** [apptainer/apptainer#2395]
### Version of Apptainer

1.3.3

### Expected behavior

I expect that if I configure apptainer source with `./mconfig -s` that the binaries will be made with static linking, since the help message includes:
```
     -s   build final host project binary statically
```

### Actual behavior

The `file` command still shows the `apptainer` and `starter` binaries are dynamically linked.  The mconfig script doesn't even attempt to change anything unless `-S` is also set, but even setting both of them makes no difference.

### Steps to reproduce this behavior

```
./mconfig -s
make -C builddir 
file builddir/apptainer builddir/cmd/starter/c/starter
```

### What OS/distro are you running

Rocky8

### How did you install Apptainer

```
git clone https://github.com/apptainer/apptainer.git
git checkout v1.3.3
```

**Comment by DrDaveD** [comment]
@cclerget said to set `CGO_ENABLED=0 CGO_LDFLAGS="-static"` when running mconfig.  I tried that and found that I also needed to use the `--without-seccomp` option.  Then when compiling with the Rocky8 golang 1.21.9 it hung.  @cclerget figured out that it did not hang with go versions downloaded from the go website, and that adding `no_openssl` to the `GO_TAGS` and `GO_TAGS_SUID` in `mlocal/frags/go_common_opts.mk` makes it also work with the Rocky8 golang rpm.  So I plan to change the `-s` option to do those things, although maybe leave it to the user to add `--without-seccomp` on hosts that need it.

I'm not sure what to do for the cross-compiling `-S` option so I would let it continue to do what was previously done when both were set.
